duffy/legacy/main.py

```diff
@@ -0,0 +1,193 @@
+from typing import Optional
+
+import httpx
+from fastapi import Depends, FastAPI
+from fastapi.responses import JSONResponse
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_401_UNAUTHORIZED,
+    HTTP_403_FORBIDDEN,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from ..version import __version__
+from .api_models import Credentials
+from .operations import obtain_credentials, obtain_destination
+
+description = """
+Duffy is the middle layer running [`ci.centos.org`](https://ci.centos.org). It provisions, tears
+down and rebuilds physical and virtual machines are used to run tests in the CentOS CI Cluster.
+
+This metaclient exposes older endpoints for legacy support and connects them to the path operations
+introduced by the newer version of the Duffy endpoint, until the support for the older endpoints is
+deprecated.
+"""
+
+app = FastAPI(
+    title="Duffy Metaclient for Legacy Support",
+    description=description,
+    version=__version__,
+    contact={"name": "CentOS CI", "email": "ci-sysadmin@centos.org"},
+)
+
+
+@app.get("/Node/get")
+async def get_node(
+    key: str,
+    ver: int = 7,
+    arch: str = "x86_64",
+    count: int = 1,
+    flavor: str = None,
+    cred: Credentials = Depends(obtain_credentials),
+):
+    if key:
+        if arch in ["aarch64", "ppc64", "ppc64le"]:
+            type = "opennebula"
+            if not flavor:
+                flavor = "medium"
+            response = httpx.post(
+                "%s/api/v1/sessions" % obtain_destination(),
+                json={
+                    "nodes_specs": [
+                        {
+                            "type": type,
+                            "quantity": count,
+                            "distro_type": "centos",
+                            "distro_version": ver,
+                            "flavour": flavor,
+                        },
+                    ],
+                },
+                auth=(cred.username, cred.password),
+            )
+        else:
+            type = "physical"
+            response = httpx.post(
+                "%s/api/v1/sessions" % obtain_destination(),
+                json={
+                    "nodes_specs": [
+                        {
+                            "type": type,
+                            "quantity": count,
+                            "distro_type": "centos",
+                            "distro_version": ver,
+                        },
+                    ],
+                },
+                auth=(cred.username, cred.password),
```

Comment:
Oh, forgot: this needs to map to pool names (which would come from say `config["metaclient"]["poolmap"]` as I described in a previous comment).